core/dbt/task/test.py

```diff
@@ -387,14 +397,24 @@ class TestTask(RunTask):
     def raise_on_first_error(self) -> bool:
         return False
 
+    @property
+    def resource_types(self) -> List[NodeType]:
+        resource_types: Collection[NodeType] = resource_types_from_args(
+            self.args, set(TEST_NODE_TYPES), set(TEST_NODE_TYPES)
+        )
+
+        # filter out any non-test node types
+        resource_types = [rt for rt in resource_types if rt in TEST_NODE_TYPES]
+        return list(resource_types)
+
     def get_node_selector(self) -> ResourceTypeSelector:
         if self.manifest is None or self.graph is None:
             raise DbtInternalError("manifest and graph must be set to get perform node selection")
         return ResourceTypeSelector(
             graph=self.graph,
             manifest=self.manifest,
             previous_state=self.previous_state,
-            resource_types=[NodeType.Test, NodeType.Unit],
+            resource_types=self.resource_types,
```

Comment:
Basically same thing as [here](https://github.com/dbt-labs/dbt-core/blob/3e437a67344469d51d7f2d4a4d0338a45c1de261/core/dbt/task/list.py#L214) and [here](https://github.com/dbt-labs/dbt-core/blob/3e437a67344469d51d7f2d4a4d0338a45c1de261/core/dbt/task/build.py#L210).